isis/src/qisis/objs/Workspace/Workspace.cpp

```diff
@@ -330,44 +330,89 @@ namespace Isis {
    *  @history 2017-10-12 Kristin Berry - Reverted to using relative instead of full file paths,
    *                          as this caused errors when working with cubelists that contained
    *                          relative paths. Fixes # 5177
+   *  @history 2018-09-12 Adam Goins - Modified logic to attempt to open the file as a cube or
+   *                          detached label first, if that fails attempt to open it as a cube list
+   *                          and if that fails, throw an error to the user. This allows cubes and
+   *                          cube lists to be saved under any extension and opened. Fixes #5439,
+   *                          Fixes #5476.
    */
-  void Workspace::addCubeViewport(QString cubename) {
-    
-    QFileInfo cubeFileName(cubename);
+  void Workspace::addCubeViewport(QString filename) {
 
-    QList<QString> cubesToOpen;
+    QFileInfo cubeFileName(filename);
+
+    QString cubename = cubeFileName.filePath();
 
-    // If the file is a cub file, we add the path to it to our list of cubes to open.
-    if ( cubeFileName.suffix() == "cub" || cubeFileName.suffix() == "cube" || cubeFileName.suffix() == "lbl") {
-      // cubesToOpen.append(cubeFileName.absoluteFilePath());
-      cubesToOpen.append(cubeFileName.filePath());
+    try {
+      Cube *cube = new Cube;
+
+      // Read in the CubeAttribueInput from the cube name
+      CubeAttributeInput inAtt(cubename);
+      std::vector<QString> bands = inAtt.bands();
+
+      // Set the virtual bands to the bands specified by the input
+      cube->setVirtualBands(bands);
+      cube->open(cubename);
+
+      MdiCubeViewport *cvp = addCubeViewport(cube);
+
+      // Check for RGB format (#R,#G,#B)
+      if(bands.size() == 3) {
+        IString st = IString(bands.at(0));
+        int index_red = st.ToInteger();
+        st = IString(bands.at(1));
+        int index_green = st.ToInteger();
+        st = IString(bands.at(2));
+        int index_blue = st.ToInteger();
+        cvp->viewRGB(index_red, index_green, index_blue);
+      }
     }
-    else {
-      // If the file received isn't a cube or label, it has to be a cubelist. We read every cube in
-      // the cubelist and append it to the cubesToOpen QList so that we can open them.
-      QFile file(cubename);
-      file.open(QIODevice::ReadOnly);
 
-      QTextStream in(&file);
+    catch (IException &e) {
+
+      QString message("Error opening cube [" + cubename + "]...\n");
+      message += "Attempting to open [" + cubename + "] as a cube list...\n";
```

Comment:
Are we sure that we want an "error" message to display if the user enters a cube list? Since a cube list is a valid option, it does not seem appropriate to ever display any kind of "error" message to the user...

Reply:
...Plus I guess this is never being displayed anyhow...

Reply:
This error message is displayed only if attempting to open it as a cube list fails and serves the purpose of alerting the user that first it was attempted to open as a cube and when that failed it attempted to open it as a cube list. If the cube list succeeds they never see it failed when it tried to open it as a cube first